internal/util/projutil/interactive_promt_util_test.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projutil
+
+import (
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+// createInput mocks the stdin input to be provided to the
+// functions by creating a temp file and writing the data to
+// it.
+func createInput(content []byte) *os.File {
+	tempFile, err := ioutil.TempFile("", "test")
+	if err != nil {
+		log.Errorf("Error while creating temp file %s", err.Error())
+	}
+	defer os.Remove(tempFile.Name())
+
+	// write content to temp file
+	if _, err := tempFile.Write(content); err != nil {
+		log.Errorf("Error while writing content %s", err.Error())
+	}
+
+	if _, err := tempFile.Seek(0, 0); err != nil {
+		log.Errorf("Error while setting the offset of the file %s", err.Error())
+	}
+
+	// Assign tempfile to os.Stdin
+	os.Stdin = tempFile
+
+	return os.Stdin
+}
+
+func bytestoString(bytes []byte) string {
+	return string(bytes[:])
+}
+
```

Comment:
Not blocker at all. I just think that would be better have one Test func with the scenarios/checks since all is testing the same func. Example:

```go

func TestGetRequiredInput(t *testing.T) {
	type args struct {
		msg string
	}
	tests := []struct {
		name string
		args args
		want string
	}{
		// TODO: Add test cases.
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := GetRequiredInput(tt.args.msg); got != tt.want {
				t.Errorf("GetRequiredInput() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

Reply:
This seems to be a good structure. Thanks for sharing! Have modified the code accordingly.